tests: Clean up InitFrameworkForRayTracingTest

sjfricke opened the pull request:
going to add supports/tests for some Ray Tracing stuff, but the `InitFrameworkForRayTracingTest` function was a little outdated and needed some refactoring and moved this out to its own PR

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 13934.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 7795 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 7795 failed.

ncesario-lunarg reviewed (commented):
These changes LGTM, but haven't investigated the CI failure yet.

@Tony-LunarG I added you as a reviewer as I _think_ you have some knowledge with this setup, but if I'm mistaken (I might be conflating with GPU-AV setup), feel free to ignore :)

Tony-LunarG commented:
Fixing InitFrameworkForRayTracingTest to call InitFramework before checking for requested extensions has resulted in several tests being run rather than being skipped.  These tests have validation errors, which causes them to fail.  In the couple I looked at, three things seem to be needed
- The buffer device address feature needs to be enabled
- Buffers used in buffer device address calls need to be created with VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT
- Memory bound to the buffer needs to be allocated with VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT_KHR
The failing tests are:
[  FAILED  ] VkLayerTest.ValidateGetRayTracingCaptureReplayShaderGroupHandlesKHR
[  FAILED  ] VkLayerTest.ValidateCreateAccelerationStructureKHR
[  FAILED  ] VkLayerTest.ValidateCreateAccelerationStructureKHRReplayFeature
[  FAILED  ] VkLayerTest.ValidateCmdTraceRaysKHR
[  FAILED  ] VkLayerTest.ValidateCmdTraceRaysIndirectKHR
[  FAILED  ] VkLayerTest.ValidateCmdBuildAccelerationStructuresKHR
[  FAILED  ] VkLayerTest.ObjInUseCmdBuildAccelerationStructureKHR
[  FAILED  ] VkLayerTest.TestCmdCopyMemoryToAccelerationStructureKHR
I patched a couple of the tests locally and those three things were what was needed to pass, but patching all the tests isn't as good as adding a method to create VkBufferObjs with VkMemoryAllocateFlagsInfo as an input.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 15298.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 7819 running.

sjfricke commented:
> Fixing InitFrameworkForRayTracingTest to call InitFramework before checking for requested extensions

I moved it back for now and added a TODO note, I don't have a good way to iterate all the test failures

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 15314.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 7820 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 7820 passed.

Tony-LunarG commented on tests/layer_validation_tests.cpp:
```diff
@@ -2473,45 +2471,44 @@ bool InitFrameworkForRayTracingTest(VkRenderFramework *renderFramework, bool isK
+        framework->AddRequiredExtensions(VK_KHR_ACCELERATION_STRUCTURE_EXTENSION_NAME);
+        framework->AddRequiredExtensions(VK_KHR_RAY_TRACING_PIPELINE_EXTENSION_NAME);
+        framework->AddRequiredExtensions(VK_KHR_RAY_QUERY_EXTENSION_NAME);
+        framework->AddRequiredExtensions(VK_KHR_GET_MEMORY_REQUIREMENTS_2_EXTENSION_NAME);
```
Memory requirements 2 is already added above

sjfricke replied:
good catch, removed

Tony-LunarG commented:
#4296 filed for skipping tests

sjfricke commented on tests/layer_validation_tests.cpp:
```diff
@@ -2473,45 +2471,44 @@ bool InitFrameworkForRayTracingTest(VkRenderFramework *renderFramework, bool isK
+    framework->InitFramework(&framework->Monitor(), enabled_features);
 
-    if (renderFramework->IsPlatform(kMockICD) || renderFramework->DeviceSimulation()) {
+    if (!mockicd_valid && (framework->IsPlatform(kMockICD) || framework->DeviceSimulation())) {
```
yes, the reason for this change is I use mockICD to tests some tests I will push after this, but decide to pull the "clean up this function" into its own PR

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 15933.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 7839 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 7839 passed.

Tony-LunarG reviewed (approved):
LGTM - thanks!
